File: 0226.Invert Binary Tree/tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        invert(root);
        return root;
    }
    void invert(TreeNode *root){
        if(root == nullptr)
            return;
        TreeNode *l = root->left, *r = root->right;
        root->left = r; root->right = l;
        invert(root->left);
        invert(root->right);
    }
};

File: offer/JZ82 二叉树中和为某一值的路径(一)/tree.java
import java.util.*;

/*
 * public class TreeNode {
 *   int val = 0;
 *   TreeNode left = null;
 *   TreeNode right = null;
 * }
 */

public class Solution {
    public boolean dfs(TreeNode root, int sum){
        if(root == null)
            return false;
        if(root.left == null && root.right == null && root.val == sum)
            return true;
        return dfs(root.left, sum - root.val) || dfs(root.right, sum - root.val);
    }
    public boolean hasPathSum (TreeNode root, int sum) {
       return dfs(root, sum);
    }
}